bundle/QueryType/SearchQueryType.php

```diff
@@ -27,38 +33,66 @@ public static function getName(): string
 
     protected function configureOptions(OptionsResolver $optionsResolver): void
     {
-        $optionsResolver->setRequired(['search_text', 'content_types', 'subtree']);
-
+        $optionsResolver->setRequired(['search_text', 'content_types', 'subtree', 'sort', 'order']);
         $optionsResolver->setAllowedTypes('search_text', 'string');
         $optionsResolver->setAllowedTypes('content_types', 'string[]');
+        $optionsResolver->setAllowedTypes('sort', 'string[]');
         $optionsResolver->setAllowedTypes('subtree', ['int', 'string']);
-
         $optionsResolver->setAllowedValues(
             'search_text',
             static fn (string $searchText): bool => trim($searchText) !== '',
         );
-
+        $optionsResolver->setAllowedValues(
+            'sort',
+            /** @var string[] $keys */
+            fn (array $keys): bool => $this->sortKeysAllowed($keys),
+        );
+        $optionsResolver->setAllowedValues(
+            'order',
+            static fn (string $order): bool => in_array($order, [Query::SORT_ASC, Query::SORT_DESC], true),
+        );
         $optionsResolver->setDefault('content_types', $this->configResolver->getParameter('search.content_types', 'ngsite'));
         $optionsResolver->setDefault('subtree', $this->site->getSettings()->rootLocationId);
+        $optionsResolver->setDefault('sort', [Query\SortClause\DatePublished::class]);
+        $optionsResolver->setDefault('order', Query::SORT_DESC);
     }
 
     protected function doGetQuery(array $parameters): Query
     {
         $subtreeLocation = $this->site->getLoadService()->loadLocation($parameters['subtree']);
-
+        $sortingKeys = $parameters['sort'];
         $criteria = [
             new Criterion\Subtree($subtreeLocation->pathString),
             new Criterion\Visibility(Criterion\Visibility::VISIBLE),
         ];
-
         if (count($parameters['content_types']) > 0) {
             $criteria[] = new Criterion\ContentTypeIdentifier($parameters['content_types']);
         }
-
         $query = new LocationQuery();
         $query->query = new FullText(trim($parameters['search_text']));
         $query->filter = new Criterion\LogicalAnd($criteria);
+        $sortClauses = [];
+        foreach ($sortingKeys as $sortingKey) {
+            $sortClauses[] = new $sortingKey($parameters['order']);
+        }
+        $query->sortClauses = $sortClauses;
 
         return $query;
     }
+
+    /**
+     * @param string[] $keys
+     *
+     * @return bool
+     */
+    private function sortKeysAllowed(array $keys): bool
```

Comment:
Can we not use `key` nomenclature here and instead use e.g. `$classNames` since this is what those are (or rather, should be)?

Reply:
@emodric  I find sortKey terminology much clearer because we know what sorting key is immediately. On the other hand, validateSortConfig(array $classNames) is ambiguous.

Perhaps sortingKeyClassName and checkSortingKeyClassNamesAllowed() (or validateSortingKeyClassNames) would be my choices at this point. 

Nevertheless,  I changed it as you instructed!